modules/geo-layers/src/tile-layer/tile-layer.ts

```diff
@@ -334,43 +342,45 @@ export default class TileLayer<DataT = any, ExtraPropsT extends {} = {}> extends
   }
 
   renderLayers(): Layer | null | LayersList {
-    return this.state.tileset.tiles.map((tile: Tile2DHeader) => {
-      const subLayerProps = this.getSubLayerPropsByTile(tile);
-      // cache the rendered layer in the tile
-      if (!tile.isLoaded && !tile.content) {
-        // nothing to show
-      } else if (!tile.layers) {
-        const layers = this.renderSubLayers({
-          ...this.props,
-          ...this.getSubLayerProps({
-            id: tile.id,
-            updateTriggers: this.props.updateTriggers
-          }),
-          data: tile.content,
-          _offset: 0,
-          tile
-        });
-        tile.layers = (flatten(layers, Boolean) as Layer<{tile?: Tile2DHeader}>[]).map(layer =>
-          layer.clone({
-            tile,
-            ...subLayerProps
-          })
-        );
-      } else if (
-        subLayerProps &&
-        tile.layers[0] &&
-        Object.keys(subLayerProps).some(
-          propName => tile.layers![0].props[propName] !== subLayerProps[propName]
-        )
-      ) {
-        tile.layers = tile.layers.map(layer => layer.clone(subLayerProps));
-      }
-      return tile.layers;
-    });
+    return (
+      this.state.tileset?.tiles.map((tile: Tile2DHeader) => {
```

Comment:
This is changing behavior, I would keep `tileset!`

Reply:
I guess we can assume its defined by the time it's called for now. If it's not defined though I wonder if returning `null` is a better behavior?

Reply:
Maybe, but I think it's best to keep the scope of this PR only to adding types. If you want to change the behavior so that `null` can be returned I think it would be beneficial to include a test case to demonstrate when something like that could occur. If it should never happen, I think having a crash here is good as it will potentially surface a bug in the future